openapi_python_client/config.py

```diff
@@ -6,37 +6,23 @@
 
 
 class ClassOverride(BaseModel):
-    class_name: str
-    module_name: str
+    class_name: Optional[str] = None
+    module_name: Optional[str] = None
 
 
 class Config(BaseModel):
-    class_overrides: Optional[Dict[str, ClassOverride]]
+    class_overrides: Dict[str, ClassOverride] = {}
     project_name_override: Optional[str]
     package_name_override: Optional[str]
     package_version_override: Optional[str]
-    field_prefix: Optional[str]
-
-    def load_config(self) -> None:
-        """ Sets globals based on Config """
-        from openapi_python_client import Project
-
-        from . import utils
-        from .parser import reference
-
-        if self.class_overrides is not None:
-            for class_name, class_data in self.class_overrides.items():
-                reference.class_overrides[class_name] = reference.Reference(**dict(class_data))
-
-        Project.project_name_override = self.project_name_override
-        Project.package_name_override = self.package_name_override
-        Project.package_version_override = self.package_version_override
-
-        if self.field_prefix is not None:
-            utils.FIELD_PREFIX = self.field_prefix
+    field_prefix: str = "field_"
 
     @staticmethod
-    def load_from_path(path: Path) -> None:
+    def load_from_path(path: Path) -> "Config":
         """ Creates a Config from provided JSON or YAML file and sets a bunch of globals from it """
+        from . import utils
+
         config_data = yaml.safe_load(path.read_text())
-        Config(**config_data).load_config()
+        config = Config(**config_data)
+        utils.FIELD_PREFIX = config.field_prefix
```

Comment:
If the `FIELD_PREFIX` is global, is there a reason all the other config isn't global too? (or vice versa, should the `FIELD_PREFIX` be part of the config)

Reply:
`FIELD_PREFIX` should be part of config eventually—I want to eliminate all globals—but when I tried to do that refactor it ended up being a ton of extra noise. Since it isn't as closely related to the ref stuff I figure I'll save it for a follow-up PR.